syntax = "proto2";

import "common/scorer.proto";

package framework;

message SiteToMineProto {
  optional int32 distance = 1;
}

message SiteProto {
  optional int32 id = 1;
  repeated SiteToMineProto deprecated_to_mine = 2;  // DistanceMap in Scorer.
}

message RiverProto {
  optional int32 source = 1;
  optional int32 target = 2;
  optional int32 punter = 3;
  optional int32 option_punter = 4;
}

message MineProto {
  optional int32 site = 1;
}

message GameMapProto {
  repeated SiteProto sites = 1;
  repeated RiverProto rivers = 2;
  repeated MineProto mines = 3;
}

message FutureProto {
  optional int32 mine_index = 1;
  optional int32 target = 2;
}

message GameStateProto {
  optional int32 punter_id = 1;
  optional int32 num_punters = 2;
  optional GameMapProto game_map = 3;
  optional common.ScorerProto scorer = 4;
  optional bool can_splurge = 5;
  optional bool can_option = 6;

  optional int32 pass_count = 7;
  optional int32 options_remaining = 8;
  optional int32 num_remaining_turns = 9;

  repeated FutureProto futures = 10;

  extensions 100 to 199;
}
